### PR TITLE
[DistillationTrainer refactor] Emit `completion_mask` alongside `labels`

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 18 (Group B) — stacked on PR 17. Part of #6449.*

First step of migrating the tensor layout to GRPO's keys (add → switch → delete). `_build_sequence_batch` now also returns the `completion_mask` it already computed internally, and both generation callers store it in the batch. Nothing consumes it yet — the loss switches to it at item 19.

- Full-sequence mask, `int` (GRPO's convention), equal to `labels != -100`.
- Buffer-side emission is throwaway (replaced by GRPO generation at item 29); the permanent part is the loss-side consumption to come.

New test `test_generated_batch_emits_completion_mask` asserts `completion_mask == (labels != -100)`.

Verified: 52 tests pass (distillation + server); ruff clean.

---

> [!NOTE]
> **Low Risk**
> Additive batch field only; loss and training semantics are unchanged until completion_mask is wired into the loss path.
> 
> **Overview**
> **DistillationTrainer** now adds a **`completion_mask`** field to on-policy batches produced after student generation, as part of aligning tensor keys with GRPO.
> 
> `_build_sequence_batch` returns the completion positions it already computed internally as a full-sequence **`int`** mask (GRPO-style), equivalent to **`labels != -100`**. Both **`model.generate`** and vLLM buffer paths attach **`completion_mask`** next to **`labels`**; **loss still masks via labels**—nothing reads **`completion_mask`** until a follow-up change.
> 
> A new integration test **`test_generated_batch_emits_completion_mask`** captures inputs at **`compute_loss`** and asserts the mask matches **`labels != -100`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a33b2897366374ac795d08496c2310359a4ad02c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_trainer.py [hunk 1]
```diff
@@ -873,7 +875,7 @@ def _build_sequence_batch(
         new_labels = torch.full_like(new_input_ids, -100)
         new_labels[completion_mask] = new_input_ids[completion_mask]
 
-        return new_attention_mask, new_labels
+        return new_attention_mask, new_labels, completion_mask.int()
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_trainer.py:878]: no that's intentional, we don't touch iw_opd_trainer.py

**Review by cmpatino — APPROVED** [review]
LGTM!
